Is filt

Redmar-van-den-Berg opened the pull request:
I have implemented the convenience property is_filt for both _Record and _Call, to indicate whether a variant or single call has been filtered. 

I often work with filtered vcf files, and so I found myself writing a bit of code to check whether a call or variant had been filtered. Like is_snp or is_indel, I figured is_filt would be useful to have in PyVCF itself.

martijnvermaat commented on vcf/model.py:
```diff
@@ -117,6 +117,22 @@ def is_het(self):
         return self.gt_type == 1
 
+    @property
+    def is_filt(self):
```
I don't think we have to abbreviate this and feel that `is_filtered` might be a better name, what do you think?

martijnvermaat commented on vcf/model.py:
```diff
@@ -117,6 +117,22 @@ def is_het(self):
+    def is_filt(self):
+        """ Return True for filtered calls """
+        try: # no FT annotation present for this variant
+            FT=self.data.FT
```
Could you [PEP 8](https://www.python.org/dev/peps/pep-0008/)-ify this? Preferably no uppercase variable names and spacing around the `=`.

martijnvermaat commented on vcf/model.py:
```diff
@@ -117,6 +117,22 @@ def is_het(self):
+            FT=self.data.FT
+        except AttributeError:
+            return False
+        if FT == None or FT == []: # FT is not set or set to PASS
```
Compare to `None` with `is` (not `==`) and check for the empty list with `len(FT) == 0`.

martijnvermaat commented on vcf/model.py:
```diff
@@ -117,6 +117,22 @@ def is_het(self):
+            return False
+        if FT == None or FT == []: # FT is not set or set to PASS
+            return False
+        elif len(FT) > 0: # FT contains one or more filters
```
Shouldn't this case already be taken care of by the one above?

martijnvermaat commented on vcf/model.py:
```diff
@@ -117,6 +117,22 @@ def is_het(self):
+            return False
+        elif len(FT) > 0: # FT contains one or more filters
+            return True
+        else: # This should not happen
```
I don't think it is needed to add special casing for things that can never happen. I'd just have a default `return True` or `return False` as the default case.

martijnvermaat commented:
Thanks! I have added a few stylistic liner notes. In general agree with the utility of this.

Redmar-van-den-Berg commented:
Thanks for your feedback. I agree that is_filtered is a better name, so I have modified this for both _Record and _Call. I have also made the other changes you suggested (PEP-8, and I've removed the trailing else clauses)

martijnvermaat commented:
Thank you!
